src/openforms/formio/utils.py

```diff
@@ -318,6 +318,11 @@ def is_visible_in_frontend(component: Component, data: DataMapping) -> bool:
     trigger_component_value = glom(data, trigger_component_key, default=None)
     compare_value = conditional.get("eq")
 
+    # Issue #2900 - The values of checkboxes are bools, but in the Formio frontend logic, the compare value is a
+    # string ('true'/'false'). In Formio they cast it to a bool.
+    if isinstance(trigger_component_value, bool):
+        compare_value = compare_value == "true"
```

Comment:
for readability sake, I think this should go under the next if clause because that one is doing a similar isinstance check, but with dict - so these two if branches now look like they can both be executed, but that's not the case! Only one can ever match.

The compare value can also be `false` to display/hide something when the checkbox is unchecked - which is confusing me in reading this line. So, maybe we can make this a bit more easy to reason about with something like:

```py
compare_value = {"true": True, "false": False}.get(compare_value, False)
```
which imo highlights the nature of mapping string booleans to actual booleans.